snapcraft/internal/repo/snaps.py

```diff
@@ -0,0 +1,214 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import contextlib
+import logging
+import sys
+import urllib
+from subprocess import check_call, check_output, CalledProcessError
+
+import requests_unixsocket
+from requests import exceptions
+
+from . import errors
+
+
+logger = logging.getLogger(__name__)
+_SNAPD_URI_ROOT_TEMPLATE = 'http+unix://%2Frun%2Fsnapd.socket/v2/{}'
+
+
+class SnapPackage:
+    """SnapPackage acts as a mediator to install or refresh a snap.
+
+    It uses information provided by snapd implicitly referring to the local
+    and remote stores to obtain information about the snap, such as its
+    confinment value and channel availability.
+
+    This information can also be used determine if a snap should be
+    installed or refreshed.
```

Comment:
There should be a "to" in here: "This information can also be used _to_ determine[...]"